yt/frontends/amrvac/data_structures.py

```diff
@@ -247,81 +242,50 @@ def _set_code_unit_attributes(self):
         # This is the reason why it uses setdefaultattr: it will only fill the gaps lefts by the "override",
         # instead of overriding them again
 
-        # conversion constants
-        He_abundance = 0.1  # hardcoded in AMRVAC
-        if self._usr_unit_system == 'mks':
-            mp = mass_hydrogen_cgs.value * 1e-3
-            kb = boltzmann_constant_cgs.value * 1e-7
-            mu0 = mu_0.value
-        else:
-            mp = mass_hydrogen_cgs.value
-            kb = boltzmann_constant_cgs.value
-            mu0 = 4*np.pi
-
         # First check if overrides have been supplied, if that's the case use those instead.
-        # note: unit_length, unit_velocity and unit_temperature are passed to Dataset._override_code_units(), but
-        # unit_numberdensity is not. This is handled here, together with a similar mylog.info() message.
-        # Retrieve them from self.units_override as (val, unit) tuple to be consistent with YT.
-        if bool(self.units_override):
-            if self._usr_unit_system == 'mks':
-                unit_length = self.units_override.get('length_unit', (1, 'm'))
-                unit_numberdensity = self.units_override.get('numberdensity_unit', (1, 'm**-3'))
-                unit_velocity = self.units_override.get('velocity_unit', (0, 'm*s**-1'))
-            else:
-                unit_length = self.units_override.get('length_unit', (1, 'cm'))
-                unit_numberdensity = self.units_override.get('numberdensity_unit', (1, 'cm**-3'))
-                unit_velocity = self.units_override.get('velocity_unit', (0, 'cm*s**-1'))
+        # Assume cgs values and let YT handle conversion if supplied in an 'mks' unit system.
+        if self.units_override:
+            unit_length = self.units_override.get('length_unit', (1, 'cm'))
+            unit_numberdensity = self.units_override.get('numberdensity_unit', (1, 'cm**-3'))
+            unit_velocity = self.units_override.get('velocity_unit', (0, 'cm*s**-1'))
```

Comment:
ok this is much better but I still think this is not necessary and misplaced.

see the order in which functions get called during a dataset initialisation ?

(from yt/data_objects/static_output.py)
```python
    def set_code_units(self):
        # here we override units, if overrides have been provided.
        self._override_code_units()

        # set attributes like ds.length_unit
        self._set_code_unit_attributes()
```

so at best this block is useless, and at worst it messes up what yt tries to do